common/src/main/java/org/mcaccess/minecraftaccess/features/point_of_interest/LockingHandler.java

```diff
@@ -286,47 +286,25 @@ public boolean lockOnEntity(Entity entity) {
     }
 
     private void findAndLockOnNearestBlock() {
-        Double minPlayerDistance = Double.MAX_VALUE;
-        Vec3d nearestBlockPosition = null;
-
-        List<TreeMap<Double, Vec3d>> scannedBlockMaps = POIBlocks.getInstance().getLockingCandidates();
-        for (TreeMap<Double, Vec3d> map : scannedBlockMaps) {
-            if (!map.isEmpty()) {
-                Entry<Double, Vec3d> closestOneInThisType = map.firstEntry();
-                Double distanceWithPlayer = closestOneInThisType.getKey();
-                if (distanceWithPlayer < minPlayerDistance) {
-                    minPlayerDistance = distanceWithPlayer;
-                    nearestBlockPosition = closestOneInThisType.getValue();
-                }
-            }
-        }
-
-        if (nearestBlockPosition != null) {
-            lockOnBlock(nearestBlockPosition);
-        }
+        POIBlocks.getINSTANCE().getLockingCandidates().stream()
+                .min(Comparator.comparingDouble(a -> WorldUtils.getClientPlayer().getEyePos().distanceTo(a.toCenterPos())))
+                .ifPresent(this::lockOnBlock);
     }
 
-    private void lockOnBlock(Vec3d position) {
+    private void lockOnBlock(BlockPos position) {
         unlock(false);
 
-        BlockState blockState = WorldUtils.getClientWorld().getBlockState(WorldUtils.blockPosOf(position));
+        BlockState blockState = WorldUtils.getClientWorld().getBlockState(position);
         entriesOfLockedOnBlock = blockState.getEntries();
```

Comment:
So... the [WorldUtils.blockPosOf()](https://github.com/minecraft-access/minecraft-access/blob/91d139b94900e5dc82a8c9a7e50e4572787830d5/common/src/main/java/org/mcaccess/minecraftaccess/utils/WorldUtils.java#L20) is a fix for breaking changes of original code. https://github.com/minecraft-access/minecraft-access/commit/ed6f890d27484304cb4aad7a117216a1ede9a384
Make sure the locking feature on things like door or button is accurate.

Reply:
Yes, I tested partial blocks and they all worked.